proxy/coco/src/state.rs

```diff
@@ -0,0 +1,733 @@
+        owner: &User,
+        project: &project::Create<P>,
+    ) -> Result<librad_project::Project<entity::Draft>, Error> {
+        let storage = self.api.storage().reopen()?;
+
+        let mut meta = project.build(owner, signer.public_key().into())?;
+        meta.sign_owned(signer)?;
+
+        let urn = meta.urn();
+        if storage.has_urn(&urn)? {
+            return Err(Error::EntityExists(urn));
+        }
+
+        let repo = storage.create_repo(&meta)?;
+        repo.set_rad_self(librad::git::storage::RadSelfSpec::Urn(owner.urn()))?;
+        log::debug!("Created project with Urn '{}'", urn);
+
+        let repo = project.setup_repo(LocalUrl::from_urn(urn, self.api.peer_id().clone()))?;
+        log::debug!("Setup repository at path '{}'", repo.path().display());
+
+        Ok(meta)
+    }
+
+    /// Create a [`user::User`] with the provided `handle`. This assumes that you are creating a
+    /// user that uses the secret key the `PeerApi` was configured with.
+    ///
+    /// # Errors
+    ///
+    /// Will error if:
+    ///     * The signing of the user metadata fails.
+    ///     * The interaction with `librad` [`librad::git::storage::Storage`] fails.
+    pub fn init_user(
+        &self,
+        signer: &signer::BoxedSigner,
+        handle: &str,
+    ) -> Result<user::User<entity::Draft>, Error> {
+        // Create the project meta
+        let mut user =
+            user::User::<entity::Draft>::create(handle.to_string(), signer.public_key().into())?;
+        user.sign_owned(signer)?;
+        let urn = user.urn();
+
+        // Initialising user in the storage.
+        {
+            let storage = self.api.storage().reopen()?;
+
+            if storage.has_urn(&urn)? {
+                return Err(Error::EntityExists(urn));
+            } else {
+                let _repo = storage.create_repo(&user)?;
+            }
+        }
+
+        Ok(user)
+    }
+
+    /// Query the network for providers of the given [`RadUrn`] within a given `timeout`.
+    pub fn providers(
+        &self,
+        urn: RadUrn,
+        timeout: Duration,
+    ) -> impl Future<Output = impl futures::Stream<Item = PeerInfo<IpAddr>>> {
+        self.api.providers(urn, timeout)
+    }
+
+    /// Wrapper around the storage track.
+    ///
+    /// # Errors
+    ///
+    /// * When the storage operation fails.
+    pub fn track(&self, urn: &RadUrn, remote: &PeerId) -> Result<(), Error> {
+        Ok(self.api.storage().track(urn, remote)?)
+    }
+
+    /// Get the [`user::User`]s that are tracking this project, including their [`PeerId`].
+    ///
+    /// # Errors
+    ///
+    /// * If we could not acquire the lock
+    /// * If we could not open the storage
+    /// * If did not have the `urn` in storage
+    /// * If we could not fetch the tracked peers
+    /// * If we could not get the `rad/self` of the peer
+    pub fn tracked(&self, urn: &RadUrn) -> Result<Vec<(PeerId, user::User<entity::Draft>)>, Error> {
```

Comment:
`nit`: "tracked" suggests a boolean check, thus I suggest renaming it to:

```suggestion
    pub fn trackers(&self, urn: &RadUrn) -> Result<Vec<(PeerId, user::User<entity::Draft>)>, Error> {
```

Reply:
#890